libs/base/ui/structure/collapsible/collapsible.component.ts

```diff
@@ -67,8 +77,26 @@ export class CollapsibleComponent
     }
   }
 
-  protected onClick(): void {
-    this.isManuallyOpened = true;
+  protected syncState(store = false): void {
+    if (!this.persistedStateKey) return;
+
+    const uiStorageKey = 'ui';
+    const uiState = JSON.parse(sessionStorage.getItem(uiStorageKey) ?? '{}');
+
+    const collapsibleStateKey = 'collapsible';
+    const collapsibleState = uiState[collapsibleStateKey] || {};
+
+    if (store) {
+      if (this.isManuallyOpened) {
+        collapsibleState[this.persistedStateKey] = this.open;
+        uiState[collapsibleStateKey] = collapsibleState;
+        sessionStorage.setItem(uiStorageKey, JSON.stringify(uiState));
```

Comment:
this can still be undefined or throw errors in some ohter cases than server side (which we did check for)